scripts/slurmsync.py

```diff
@@ -519,8 +519,7 @@ def main():
     with pid_file.open("w") as fp:
         try:
             fcntl.lockf(fp, fcntl.LOCK_EX | fcntl.LOCK_NB)
+            main()
```

Comment:
Moving it here breaks `force` logic. `main` will no be executed if exception raises (no matter what type or `args.force`).
Can we achieve the desired change by simply indenting `main()` into `with` block ?

Reply:
Thanks for catching that. 

I made the force branch separate to let it succeed also when it can not for whatever reason open the lock file with  in writable mode (letting --force really mean force)

I also updated the PR and also narrowed the except statement to more narrowly catch only failure to acquire lock.